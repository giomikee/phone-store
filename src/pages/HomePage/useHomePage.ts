import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import type { Phone } from '../../interfaces';
import { useEffect, useRef, useState } from 'react';
import { INITIAL_PHONES_COUNT, PAGES } from '../../constants';
import { getPhones } from '../../services';
import { addPhones } from '../../state/phone/phone.slice';
import { useStoreSelectors } from '../../state/useStoreSelectors';

export function useHomePage() {
    const navigate = useNavigate();

    const {
        phones,
        searchedPhones,
        areSearchResultsDisplayed,
    } = useStoreSelectors();

    const dispatch = useDispatch();

    const [isLoading, setIsLoading] = useState(false);
    const [isLoadingSearch, setIsLoadingSearch] = useState(false);
    const [isInitialLoad, setIsInitialLoad] = useState(true);
    const hasFetchedRef = useRef(false);  // Avoid calling useEffect twice in dev
    let visiblePhones: Phone[];


    if (areSearchResultsDisplayed) {
        visiblePhones = searchedPhones;
    } else {
        visiblePhones = isInitialLoad ? phones.slice(0, INITIAL_PHONES_COUNT) : phones;
    }

    const fetchPhones = async () => {
        try {
            setIsLoading(true);
            const fetchedPhones = await getPhones({ limit: INITIAL_PHONES_COUNT });

            dispatch(addPhones(fetchedPhones));
        } catch (error) {
            console.error(error);

            navigate(PAGES.error);
        } finally {
            setIsLoading(false);
        }
    };

    useEffect(() => {
        if (phones.length > 0 || hasFetchedRef.current) {
            return;
        }

        hasFetchedRef.current = true;

        fetchPhones();
    }, []);

    return {
        isLoading,
        isInitialLoad,
        isLoadingSearch,
        visiblePhones,
        hasFetchedRef,
        fetchPhones,
        setIsLoadingSearch,
        setIsInitialLoad,
    };
}