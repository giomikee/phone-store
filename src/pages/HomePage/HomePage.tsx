import { Container, Divider, Grid, Skeleton, Typography } from '@mui/material';
import { HEADINGS, INITIAL_PHONES_COUNT, PHONES_GRID } from '../../constants';
import LoadAllPhones from '../../components/LoadAllPhones/LoadAllPhones';
import SearchPhonesInput from '../../components/SearchPhonesInput/SearchPhonesInput';
import PhoneCards from '../../components/PhoneCards/PhoneCards';
import { useHomePage } from './useHomePage';
import { useStoreSelectors } from '../../state/useStoreSelectors';

const HomePage = () => {
    const { areSearchResultsDisplayed } = useStoreSelectors();
    const {
        isLoading,
        isInitialLoad,
        isLoadingSearch,
        visiblePhones,
        setIsLoadingSearch,
        setIsInitialLoad,
    }= useHomePage();

    return (
        <Container>
            <Typography 
                component={HEADINGS.h2}
                variant={HEADINGS.h5}
            >
                Buscar MÃ³viles
            </Typography>

            <Divider sx={{ my: 2 }} />

            {
                isLoading 
                ? <Skeleton
                    height={80} 
                    variant='text' />
                : <SearchPhonesInput onSearchLoading={setIsLoadingSearch} />
            }

            <Divider sx={{ my: 2 }} />

            <Grid 
                container
                spacing={2}
            >
                {
                    (isLoading || isLoadingSearch) && Array.from({ length: INITIAL_PHONES_COUNT }).map((_, index) => 
                        <Grid 
                            key={index}
                            size={PHONES_GRID}
                        >
                            <Skeleton 
                                height={400} 
                                variant='rectangular'
                            />
                        </Grid>
                    )
                }

                <PhoneCards phones={visiblePhones} />

                {
                    !isLoading && !areSearchResultsDisplayed && isInitialLoad && 
                        <LoadAllPhones onLoadAll={() => setIsInitialLoad(false)} />
                }
            </Grid>
        </Container>
    )
}

export default HomePage