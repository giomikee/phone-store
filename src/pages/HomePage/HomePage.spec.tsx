import { describe, expect, it, vi, type Mock } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import HomePage from './HomePage';
import { useHomePage } from './useHomePage';

vi.mock('./useHomePage', () =>({
    useHomePage: vi.fn().mockReturnValue({
        isLoading: false,
        isInitialLoad:true,
        isLoadingSearch: false,
        visiblePhones: [
            {
                id: 'id',
                name: 'name',
                brand: 'brand',
                basePrice: 333,
            },
        ],
        setIsLoadingSearch: vi.fn(),
        setIsInitialLoad: vi.fn(),
    })
}))

vi.mock('../../components/LoadAllPhones/useLoadAllPhones');

vi.mock('../../state/useStoreSelectors');

vi.mock('react-redux', async () => ({
    ...(await vi.importActual('react-redux')),
    useDispatch: vi.fn().mockReturnValue(vi.fn()),
}));

vi.mock('react-router-dom', async () => ({
    ...(await vi.importActual('react-router-dom')),
    useNavigate: vi.fn().mockReturnValue(vi.fn()),
}));

describe('Given HomePage component', () => { 
    const mockedUseHomePage = {
        isLoading: false,
        isInitialLoad:true,
        isLoadingSearch: false,
        areSearchResultsDisplayed: false,
        visiblePhones: [
            {
                id: 'id',
                name: 'name',
                brand: 'brand',
                basePrice: 333,
            },
        ],
        setIsLoadingSearch: vi.fn(),
        setIsInitialLoad: vi.fn(),
    };

    describe('When component is rendered', () => { 
        describe('And isLoading is true', () => { 
            it('Then it should match snapshot', () => {
                (useHomePage as Mock).mockReturnValueOnce({
                    ...mockedUseHomePage,
                    isLoading: true,
                });

                const { container} = render(<HomePage />)

                expect(container).toMatchSnapshot()
            })
        })

        describe('And isLoading is false', () => { 
            describe('But areSearchResultsDisplayed is false', () => { 
                describe('And isInitialLoad is true', () => { 
                    it('Then it should match snapshot', () => {
                        (useHomePage as Mock).mockReturnValueOnce({
                            ...mockedUseHomePage,
                            isLoading: false,
                            areSearchResultsDisplayed: false,
                            isInitialLoad: true,
                        });

                        const { container } = render(<HomePage />)

                        expect(container).toMatchSnapshot()
                    })
                })
            })
        });
    });
})