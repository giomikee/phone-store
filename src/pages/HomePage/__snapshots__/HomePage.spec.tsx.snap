// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given HomePage component > When component is rendered > And isLoading is false > But areSearchResultsDisplayed is false > And isInitialLoad is true > Then it should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
    >
      Buscar Móviles
    </h2>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-zzw4lh-MuiDivider-root"
    />
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
      data-testid="search-phones-input"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="search-phones-input"
        id="search-phones-input-label"
      >
        Buscar Móviles 
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1n04w30-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
          id="search-phones-input"
          name="search-phones-input"
          type="text"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-3nihpm-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1n64csd-MuiNotchedOutlined-root"
          >
            <span>
              Buscar Móviles 
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-zzw4lh-MuiDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-zjhuaj-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j74uew-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root css-coyn9m-MuiButtonBase-root-MuiCardActionArea-root"
            data-testid="phone-card-action-area-id_0"
            tabindex="0"
            type="button"
          >
            <img
              alt="brand name"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1kuwa16-MuiCardMedia-root"
              height="300"
            />
            <div
              class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-1rdkf6a-MuiChip-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorPrimary css-1umw9bq-MuiSvgIcon-root"
                    data-testid="SmartphoneIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 1.01 7 1c-1.1 0-2 .9-2 2v18c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V3c0-1.1-.9-1.99-2-1.99M17 19H7V5h10z"
                    />
                  </svg>
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
                  >
                    brand
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-filledSecondary css-3p8epo-MuiChip-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorSecondary css-1umw9bq-MuiSvgIcon-root"
                    data-testid="EuroIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57"
                    />
                  </svg>
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
                  >
                    desde 333.00 €
                  </span>
                </div>
              </div>
              <h3
                class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
              >
                name
              </h3>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1h5un5t-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-17ls7xt"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
          data-testid="load-all-phones-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="ExpandCircleDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11z"
              />
            </svg>
          </span>
          <span
            class="MuiButton-loadingWrapper"
            style="display: contents;"
          />
          Ver todo
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Given HomePage component > When component is rendered > And isLoading is true > Then it should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
    >
      Buscar Móviles
    </h2>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-zzw4lh-MuiDivider-root"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
      style="height: 80px;"
    />
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-zzw4lh-MuiDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-zjhuaj-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
          style="height: 400px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-19vcg0t-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j74uew-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root css-coyn9m-MuiButtonBase-root-MuiCardActionArea-root"
            data-testid="phone-card-action-area-id_0"
            tabindex="0"
            type="button"
          >
            <img
              alt="brand name"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1kuwa16-MuiCardMedia-root"
              height="300"
            />
            <div
              class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-1rdkf6a-MuiChip-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorPrimary css-1umw9bq-MuiSvgIcon-root"
                    data-testid="SmartphoneIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 1.01 7 1c-1.1 0-2 .9-2 2v18c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V3c0-1.1-.9-1.99-2-1.99M17 19H7V5h10z"
                    />
                  </svg>
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
                  >
                    brand
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-filledSecondary css-3p8epo-MuiChip-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorSecondary css-1umw9bq-MuiSvgIcon-root"
                    data-testid="EuroIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57"
                    />
                  </svg>
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
                  >
                    desde 333.00 €
                  </span>
                </div>
              </div>
              <h3
                class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
              >
                name
              </h3>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1h5un5t-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
